### PR TITLE
refactor(graphql): migrate from nexus to pothos

### LINKED ISSUE #478: migrate from nexus to pothos
to simplify dependencies.

### PR DESCRIPTION
**Pull request by PascalHelbig**
fixes #478 

Todo:
- [x] Test on windows

### CODE REVIEW

File: app/backend/graphql/project/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
-export * from './project'
-export * from './queries'
-export * from './mutations'
+import './queries'
+import './mutations'
+
+export { Project } from './project'
```
Comment (moncapitaine) [inline app/backend/graphql/project/index.ts:4]: used where?

File: app/backend/graphql/project/mutations/projectDeleteMutationField.ts [hunk 2]
```diff
@@ -1,29 +1,25 @@
...
 
-    if (project.team.slug !== context.teamSlug) {
-      // We can not delete a project from a different team
-      throw new Error('not authenticated')
-    }
-
-    return context.prisma.project.delete({ where: { id } })
-  },
-})
+      return prisma.project.delete({ where: { id: id.toString() } })
```
Comment (moncapitaine) [inline app/backend/graphql/project/mutations/projectDeleteMutationField.ts:22]: ...query is missing

File: app/backend/graphql/project/mutations/projectDeleteMutationField.ts [hunk 3]
```diff
@@ -1,29 +1,25 @@
...
+  t.withAuth({ isTeamAdmin: true }).prismaField({
+    type: 'Project',
+    description: 'Delete a project',
+    args: {
+      id: t.arg.id({ description: 'id of the project' }),
+    },
+    resolve: async (query, _source, { id }, context) => {
+      const project = await prisma.project.findUniqueOrThrow({
+        where: { id: id.toString() },
+        include: { team: true },
```
Comment (moncapitaine) [inline app/backend/graphql/project/mutations/projectDeleteMutationField.ts:14]: include the complete team? slug would do

File: app/backend/graphql/project/project.ts [hunk 4]
```diff
@@ -1,61 +1,39 @@
...
+    id: t.exposeID('id', { description: 'identifies the project' }),
+    title: t.exposeString('title'),
+    startDate: t.expose('startDate', { type: 'Date', nullable: true }),
+    endDate: t.expose('endDate', { type: 'Date', nullable: true }),
+    workHours: t.field({
+      type: [WorkHour],
+      select: { tasks: { select: { workHours: { select: { id: true } } } } },
+      resolve: (project) => project.tasks.flatMap((task) => task.workHours),
+    }),
+    // TODO: add workHours
```
Comment (moncapitaine) [inline app/backend/graphql/project/project.ts:19]: remove comment

File: app/backend/graphql/projectMembership/mutations/projectMembershipCreateMutationField.ts [hunk 5]
```diff
@@ -1,46 +1,45 @@
...
+      const teamMembership = await prisma.teamMembership.findUnique({
+        select: { id: true },
+        where: { userId_teamId: { userId: userId.toString(), teamId: project.team.id } },
+      })
+      if (!teamMembership) {
+        return false
+      }
+      return true
+    },
+    resolve: async (query, _source, { userId, projectId }, context) => {
```
Comment (moncapitaine) [inline app/backend/graphql/projectMembership/mutations/projectMembershipCreateMutationField.ts:30]: remove

File: app/backend/graphql/projectMembership/mutations/projectMembershipDeleteMutationField.ts [hunk 6]
```diff
@@ -1,44 +1,43 @@
...
+      const teamMembership = await prisma.teamMembership.findUnique({
+        select: { id: true },
+        where: { userId_teamId: { userId: userId.toString(), teamId: project.team.id } },
+      })
+      if (!teamMembership) {
+        return false
+      }
+      return true
+    },
+    resolve: async (query, _source, { userId, projectId }, context) => {
```
Comment (moncapitaine) [inline app/backend/graphql/projectMembership/mutations/projectMembershipDeleteMutationField.ts:30]: remove

File: app/backend/graphql/team/mutations/teamAcceptInviteMutationField.ts [hunk 7]
```diff
@@ -1,39 +1,38 @@
...
 
-    // Create new membership
-    return context.prisma.team.update({
-      where: { inviteKey },
-      data: {
-        teamMemberships: {
-          create: { userId: context.session.user.id },
+      // Create new membership
+      return prisma.team.update({
+        where: { inviteKey },
```
Comment (moncapitaine) [inline app/backend/graphql/team/mutations/teamAcceptInviteMutationField.ts:29]: ...query missing

**Review by moncapitaine — APPROVED** [review]

File: app/backend/graphql/project/index.ts (hunk 1 above)
Comment (PascalHelbig) [reply app/backend/graphql/project/index.ts:4]: Sometimes we have to import a pothos object type directly, e.g. we have to import `User` into `Project`.
I would keep it consistent, and always export the object types.

**Pull request merged**